packages/rtk-query-graphql-request-base-query/src/GraphqlBaseQueryTypes.ts

```diff
@@ -12,14 +13,23 @@ export type PrepareHeaders = (
   api: Pick<BaseQueryApi, 'getState' | 'endpoint' | 'type' | 'forced' | 'extra'>
 ) => MaybePromise<Headers>
 
+export type ErrorResponse = {
+  message: string;
+  stack: string;
+  name: string;
+  [key: string]: unknown;
```

Comment:
Do you have a specific reason for the index signature? I'd rather leave that one out. We could also think about adding a generic error type, but I'd suggest adding that at a later date if the need really should come up.

Reply:
Hi, I use the index signature to generalize the error response starting from the base error. And in my app, the first thing I do was add another field related to a specific requirement.

Here, for example, I add a property **errorCode**
```    
...
customErrors: ({ name, message, request, stack, response }) => {
      return {
        name,
        stack,
        message: response?.errors[0]?.extensions?.errorMessage,
        errorCode: response?.errors[0]?.extensions?.errorCode    // <---------- ADDITIONAL FIELD
      };
    }
```

But I agree with you, you can leave out the index field and on my side, i will try to find a simple and consistent way to generalize the response type

Reply:
I pushed an update

```
export type ErrorResponse = {
  message: string;
  stack: string;
  name: string;
};

export type GraphqlRequestBaseQueryArgs<E = ErrorResponse> = (
  ...
  customErrors?: (args: ClientError) => E;
}
```


And the inizialitazion of the client becomes (passing the error response type): 
```

export const api = createApi({
  baseQuery: graphqlRequestBaseQuery<{foo: number}>({
    ....
    customErrors: (args) => {
      return {foo: 1, };
    }
  }),
  ...
});
```

Or without passing error response type

```

export const api = createApi({
  baseQuery: graphqlRequestBaseQuery({
    ....
    customErrors: (args) => {
      return {m:essage: "", stack: "", name: " };
    }
  }),
  ...
});
```